### PR TITLE
feat: parametrize system prompt

### PR DESCRIPTION
**Pull request by jaideepr97**
This PR parametrizes the system prompt so that users of SDG can pass their own system prompt in to the SDG process

fixes: #336

### CODE REVIEW

**Comment by jaideepr97** [comment]
This can wait until https://github.com/instructlab/sdg/pull/284 is in

**Comment by jaideepr97** [comment]
I think we have a chicken and egg problem here - CI is likely failing because the CLI version being used for the medium and large jobs doesn't know to pass the sys prompt in as a parameter
but the CLI PR to do this will also not pass until this PR is merged into SDG and a new release is cut 

cc @nathan-weinberg

**Comment by jaideepr97** [comment]
> In a way this is a good thing, as it forces us to think about backwards compatibility of our SDG library API. For example, instead of making the new system prompt required, we could make it optional and default to the current system prompt if not passed. That would have no issue passing the current CI, and then once this gets merged and a new SDG release cut we could bump the min SDG version in the CLI repo and change it to pass in the optional system prompt parameter.

ah yeah, makes sense. I should've thought of that haha
thanks @bbrowning !

**Review by aakankshaduggal — APPROVED** [review]
Thanks @jaideepr97!

**Comment by jaideepr97** [comment]
@aakankshaduggal @bbrowning 
I'm going to update this PR to include the student_model_arch as a param that can be used to determine the correct pretraining data format for further work

**Comment by aakankshaduggal** [comment]
> @aakankshaduggal @bbrowning I'm going to update this PR to include the student_model_arch as a param that can be used to determine the correct pretraining data format for further work

Sounds good, thanks @jaideepr97 that would be amazing!

**Comment by jaideepr97** [comment]
on second thought, it made more sense to have a separate PR for it 
so I took a stab at it here: https://github.com/instructlab/sdg/pull/341

File: src/instructlab/sdg/generate_data.py
Comment (Maxusmusti) [inline src/instructlab/sdg/generate_data.py:99]: Since we are still using a default system prompt as a fallback, we probably want to update that to be the 8B prompt by default

Comment (jaideepr97) [reply src/instructlab/sdg/generate_data.py]: updated, thanks

**Review by bbrowning — COMMENTED** [review]
A nice side-effect of the last commit that updates the default system prompt to Granite 3.0 is our test is now also validating that the user can pass in a custom system prompt to `generate_data` and it actually takes effect, since the system prompt we pass in there is the older prompt. 

Assuming this passes the e2e tests, the code itself looks good to me. Thanks for centralizing the logic that guards against the `None` system prompt value as well!

I'm holding off on actual approval until we get 284 in, as it should be very close and the surface area here for resolving any merge/rebase conflicts is lower than in 284.

**Review by khaledsulayman — COMMENTED** [review]
Thanks for this, Jaideep!

Echoing Ben's held-off approval until we get #284 in. Happy to work with you to help rebase if you need.

**Review by khaledsulayman — APPROVED** [review]

**Pull request merged**
